src/utils/webAppUtil.ts

```diff
@@ -0,0 +1,90 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ApiContract, BackendContract, BackendCredentialsContract } from "azure-arm-apimanagement/lib/models";
+import WebSiteManagementClient from "azure-arm-website";
+import { Site, WebAppCollection } from "azure-arm-website/lib/models";
+import xml = require("xml");
+import { ApisTreeItem } from "../explorer/ApisTreeItem";
+import { ApiTreeItem } from "../explorer/ApiTreeItem";
+import { ext } from "../extensionVariables";
+import { azureClientUtil } from "./azureClientUtil";
+import { nonNullOrEmptyValue } from "./nonNull";
+
+export namespace webAppUtil {
+    export async function getPickedWebApp(node: ApiTreeItem | ApisTreeItem, webAppType: string): Promise<Site> {
+        const client = azureClientUtil.getClient(node.root.credentials, node.root.subscriptionId, node.root.environment);
+        const allfunctionApps = await listWebApps(client, webAppType);
+        return await pickWebApp(allfunctionApps);
+    }
+
+    export async function listWebApps(client: WebSiteManagementClient, webAppType: string): Promise<Site[]> {
```

Comment:
should this be in the service

Reply:
it is also used to list function apps, maybe we can put it in the import webapps

Reply:
also it can't be put in the service, since we need to pick the web app at first, then we can create the service